### ISSUE TITLE
clang-format poor comment spacing with AlignArrayOfStructures

### ISSUE THREAD

**Issue by kadler** [llvm/llvm-project#69342]
Similar to #62904 I've also noticed weird alignment issues of comments inside array initializers when AlignArrayOfStructures is enabled:

test.c
```c
struct point {
    int x, y;
}

struct point points[] = {
    // test of single line comment
    {0, 1},
    /* this is a C-style comment */
    {1, 2},
            // overly indented comment
    {3, 4},
        /* overly indented C-style comment */
    {0, 0}
};
```

```
clang-format -style="{ ContinuationIndentWidth: 8, AlignArrayOfStructures: Left }" test.c
```
```c
struct point {
  int x, y;
}

struct point points[] = {
  // test of single line comment
        {0, 1},
 /* this is a C-style comment */
        {1, 2},
 // overly indented comment
        {3, 4},
 /* overly indented C-style comment */
        {0, 0}
};
```
```
clang-format -style="{ ContinuationIndentWidth: 8, AlignArrayOfStructures: Right }" test.c
```
```c
struct point {
  int x, y;
}

struct point points[] = {
  // test of single line comment
        {0, 1},
 /* this is a C-style comment */
        {1, 2},
 // overly indented comment
        {3, 4},
 /* overly indented C-style comment */
        {0, 0}
};
```

```
clang-format -style="{ ContinuationIndentWidth: 8, AlignArrayOfStructures: None }" test.c
```
```c
struct point {
  int x, y;
}

struct point points[] = {
        // test of single line comment
        {0, 1},
        /* this is a C-style comment */
        {1, 2},
        // overly indented comment
        {3, 4},
        /* overly indented C-style comment */
        {0, 0}};
```

When Using AlignArrayOfStructures with either Left or Right options, the indentation of the comments does not follow the indentation of the array contents. The first comment is indented two spaces while subsequent comments are indented one space. Neither IndentWidth nor ContinuationindentWidth seem to have any effect on this behavior.

This is with clang-format 16.0.6 from macOS Homebrew.

**Comment by ptc-tgamper** [comment]
I hit this one, too. clang-format 16.0.6.

**Comment by EugeneZelenko** [comment]
Could you please try 17 or `main` branch?

**Comment by ptc-tgamper** [comment]
Still happens on 17.0.6

**Comment by XDeme** [comment]
Couldn't reproduce it with 
```
❯ clang-format --version
clang-format version 18.0.0 (https://github.com/llvm/llvm-project.git a6c02edd6eac476523b5c73f29619a7a9e054872)
```
This is probably fixed.

**Comment by rymiel** [comment]
This was fixed by fff993b7cfd01f733216bd81d2a18838b7a4a97f
